db/database.go

```diff
@@ -330,24 +330,30 @@ func (context *DatabaseContext) NotifyUser(username string) {
 }
 
 func (dc *DatabaseContext) TakeDbOffline(reason string) error {
-	base.LogTo("CRUD", "Taking Database : %v, offline", dc.Name)
+
 	dbState := atomic.LoadUint32(&dc.State)
+
+	// TODO: if we're taking the bucket offline, why aren't we closing the DCP feed?
+	// TODO: why aren't we calling dc.Close() here?  Where does it get called?
+
+	base.LogTo("CRUD", "Taking Database : %v, offline.  dbState: %v", dc.Name, dbState)
```

Comment:
I think we worked out that this call was just about changing the database state (with the accompanying change to the REST API behaviour), and not about actually closing the DB.  This allows it to be called from a variety of circumstances.